functions/go/templater/function/filter_test.go

```diff
@@ -0,0 +1,206 @@
+package function
+
+import (
+	"testing"
+
+	"bytes"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+func TestTemplates(t *testing.T) {
+
+	os.Setenv("TESTTEMPLATE", "testtemplatevalue")
+
+	tc := []struct {
+		cfg         string
+		in          string
+		expectedOut string
+		expectedErr bool
+	}{
+		{
+			cfg: `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: notImportantHere
+data:
+  template: |
+    value1: {{ .literal1 }}
+  literal1: value1
+  literal2: value2
+`,
+			expectedOut: `value1: value1
+`,
+		},
+		{
+			cfg: `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: notImportantHere
+data:
+  template: 'value: {{ env "TESTTEMPLATE" }}'
+`,
+			expectedOut: `value: testtemplatevalue
+`,
+		},
+		{
+			cfg: `
+apiVersion: v1alpha1
+kind: Templater
+metadata:
+  name: notImportantHere
+data:
+  template: |
+    {{ range .hosts -}}
+    ---
```

Comment:
this `---` is needed because the intent was to generate a yaml document for each objects in `.hosts`. 
kio.Pipeline ignores empty documents, when reads and converts buffer to RNode [here](https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/103/files/84f2c5702d021f925437633b9db442486dcaf86c#diff-a281396d074fd7355fe0e872747be5e1R63).
that's why we see only 2 generated documents.
I could put `---` near `{{ end -}}`, the result should be the same.

Reply:
I see thanks!